core/commonMain/src/implementations/persistentOrderedMap/PersistentOrderedMapBuilderContentIterators.kt

```diff
@@ -28,7 +28,10 @@ internal open class PersistentOrderedMapBuilderLinksIterator<K, V>(
         lastIteratedKey = nextKey
         nextWasInvoked = true
         index++
-        val result = builder.hashMapBuilder[nextKey]!!
+        @Suppress("UNCHECKED_CAST")
+        val result = builder.hashMapBuilder.getOrElse(nextKey as K) {
```

Comment:
Observation: the unchecked cast is the result of an inconsistency between `get` and `getOrElse` functions for Map. Perhaps, it's worth addressing that in stdlib.